src/pages/AssignServerRoles/BaremetalSettings.js

```diff
@@ -0,0 +1,93 @@
+// (c) Copyright 2017 SUSE LLC
+/**
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React, { Component } from 'react';
+import { translate } from '../../localization/localize.js';
+import { ConfirmModal } from '../../components/Modals.js';
+import { IpV4AddressValidator } from '../../utils/InputValidators.js';
+import { ServerInputLine } from '../../components/ServerUtils.js';
+import { ActionButton } from '../../components/Buttons.js';
+
+class BaremetalSettings extends Component {
+
+  constructor(props) {
+    super(props);
+    const baremetal = this.props.model.getIn(['inputModel', 'baremetal']).toJS()
+    this.state = {
+      subnet: baremetal.subnet,
+      netmask: baremetal.netmask
+    }
+    this.origBaremetal = JSON.parse(JSON.stringify(baremetal));
+  }
+
+  handleInputChange = (e, valid, props) => {
```

Comment:
When I put in an invalid subnet (e.g. 1.2.3.4.5), the field validator is invoked and shows red (Please enter a valid IPV4 address), but the Save button is still enabled, and it will actually save a portion of the value that I entered (which is clearly not correct). If I put in a value like 'suse', it is also permitted but seems to do nothing. The Save button should be disabled when the values are not valid